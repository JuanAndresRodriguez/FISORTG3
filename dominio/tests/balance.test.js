const { TestWatcher } = require('@jest/core');
const Balance = require('../balance');
const Income = require('../income');
const Expense = require('../expense');

let cuenta = new Balance();
let sueldo = new Income("Sueldo",600,"2021-10-07","UYU",true);
let renner = new Expense("renner",200,"2021-10-09","UYU","Ropa",true);

test('add income to balance',() => {
    cuenta.addIncomeToBalance(sueldo.getIncome());
    expect(cuenta.balanceMoney).toBe(600)
});
test('add expense to balance',() =>{
    cuenta.addExpenseToBalance(renner.getExpense());
    expect(cuenta.balanceMoney).toBe(400);
    
});
test('check income array list',()=>{
    expect(cuenta.incomeList.length).toBe(1)
});
test('check expense array list',()=>{
    expect(cuenta.expensesList.length).toBe(1)
});
