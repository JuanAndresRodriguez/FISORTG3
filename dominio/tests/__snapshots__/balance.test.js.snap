// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Balance {
  "balanceMoney": 400,
  "expensesList": Array [
    Object {
      "amount": 200,
      "category": "Ropa",
      "currency": "UYU",
      "date": 2021-10-09T00:00:00.000Z,
      "isMonthly": true,
      "name": "renner",
    },
  ],
  "incomeList": Array [
    Object {
      "amount": 600,
      "currency": "UYU",
      "date": 2021-10-07T00:00:00.000Z,
      "isMonthly": true,
      "name": "Sueldo",
    },
  ],
}
`;
